redhat-appstudio/service-provider-integration-operator
# make tokenuploader create new SPIAccessToken

### What does this PR do?
Creates new SPIAccessToken, associates it with uploaded with Secret one and moves SPIAccessToken to Ready state.

### Screenshot/screencast of this PR

n/a

### What issues does this PR fix or reference?
Makes it possible to not only associate token with existed SPIaccessToken but create and make them ready to use right away. It simplifies the flow for Admin/DevOPs who need to create simple secret (for example by initalization script) but do not use it, so not needs SPIAccessTokenBinding. 

https://issues.redhat.com/browse/SVPI-382


### How to test this PR?
There is dedicated integration test included, to run it call `make itest focus="TokenUploadController"`

To reproduce it manually
- make sure SPIOperator configured with `ENABLETOKENUPLOAD=true` (see Admin Guide for details)
- Obtain Access Token ($AT) from the Service Provider of your choice. For example from GitHub->Settings->Developer Settings->Personal Access Tokens  
- Create Secret l: `kubectl apply -f my.yaml` 
With `my.yaml `file content like: 
` 
```
apiVersion: v1
kind: Secret
metadata:
  name: upload-secret
  labels:
    spi.appstudio.redhat.com/upload-secret: token
    spi.appstudio.redhat.com/token-name: my-spi-access-token
type: Opaque
stringData:
  providerUrl: https://github.com
  tokenData: $AT
```
`
- Make sure SPIAccessToken is created and Ready state (may take some time): `kubectl get spiaccesstoken my-spi-access-token -o yaml `
(status:phase:Ready)
